docs/DesignDocs/oe_log_message()_callback_proposal.md

```diff
@@ -0,0 +1,61 @@
+Proposal of Public Callback API for oe_log_message()
+====
+
+# Motivation
+
+This document describes the proposed public callback API for oe_log_message().
+
+There are several problems about the current logging that prevent end users to
+make full use of log:
+- It does't have mechanism to communicate between enclave and host.
+- Message can't be redirected to different destinations on demand.
+- Lack of method for end user to customize messages.
+
+The callback API is supposed to solve such problems.
+
+# User Experience
+
```

Comment:
It's not clear to me how the above three problems are addressed based on the current context. It would be good to re-structure a bit.

Reply:
I have added some sentence and also a sample, thanks!